interpreter/binary/decode.ml

```diff
@@ -355,6 +350,20 @@ let locals s =
     s pos "too many locals";
   List.flatten (List.map (Lib.Fun.uncurry Lib.List32.make) nts)
 
+let on_clause s =
+  if peek s = Some 0x01 then begin
```

Comment:
Can we avoid peek when not necessary? Just match on the result of get.

Reply:
Yes, done!